assets/js/dependencies/async_loader.js

```diff
@@ -1,48 +1,250 @@
+          if(installed!=false){
+            var examples= $('.topic').find('.topic--title').filter(function(i,el){
+              return $(this).text()=="Examples";
+            }).parent().find('.R').text();
+            _rStudioRequest('/rpc/console_input','console_input',urlParam("RS_SHARED_SECRET"),urlParam("Rstudio_port"),["require("+package+")\n"+examples])
+          }
+        });
+      };
 
-    function rerenderBody(html){
-      var body = html.replace(/^[\S\s]*<body[^>]*?>/i, "").replace(/<\/body[\S\s]*$/i, "");
-      $pageBody.html(body);
-      bindGlobalClickHandler();
-      window.searchHandler(jQuery);
-      window.packageVersionToggleHandler(jQuery);
-      MathJax.Hub.Queue(["Typeset",MathJax.Hub])
+      window.setDefault=function(){
+        _rStudioRequest('/rpc/console_input','console_input',urlParam("RS_SHARED_SECRET"),urlParam("Rstudio_port"),["Rdocumentation::makeDefault()"]);
+      }
+      window.hideViewer=function(){
+        _rStudioRequest('/rpc/console_input','console_input',urlParam("RS_SHARED_SECRET"),urlParam("Rstudio_port"),["Rdocumentation::hideViewer()"]);
+      }
     }
 
-    // Helper function to grab new HTML
-    // and replace the content
-    window.replacePage = function(url) {
-      $.ajax({
-        type: 'GET',
-        url: 'https://rdocs-v2.herokuapp.com' + url + '?viewer_pane=1',
-        cache: false,
-        dataType: 'html',
-        xhrFields: {
-          withCredentials: true
+
+  });
+
+_rStudioRequest=function(url,method,shared_secret,port,params){
+  var data={}
+  data["method"]=method;
+  //data["params"]=[$('.R').text()];
+  data["params"]=params;
+  data["clientId"]='33e600bb-c1b1-46bf-b562-ab5cba070b0e';
+  data["clientVersion"]="";
+  return $.ajax({
+    url: 'http://127.0.0.1:'+port+url,
+    headers:
+    {
+        'Accept':'application/json',
+        'Content-Type':'application/json',
+        'X-Shared-Secret':shared_secret
+    },
+    type: 'POST',
+    dataType: 'json',
+    data: JSON.stringify(data),
+    processData: false,
+    crossDomain:true,
+    xhrFields: {
+      withCredentials: true
+    }
+  });
+}
+_versionCompare = function (v1, v2) {
+    v1parts = v1.split(/[.-]+/); 
+    v2parts = v2.split(/[.-]+/); 
+    while (v1parts.length < v2parts.length) v1parts.push("0");
+    while (v2parts.length < v1parts.length) v2parts.push("0");
+    v1parts = v1parts.map(Number);
+    v2parts = v2parts.map(Number);
+    for (var i = 0; i < v1parts.length; ++i) {
+        if (v2parts.length == i) {
+            return true;
         }
-      })
-      .done(rerenderBody);
-    };
 
-    bindGlobalClickHandler();
-  }
+        if (v1parts[i] == v2parts[i]) {
+            continue;
+        }
+        else if (v1parts[i] > v2parts[i]) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
 
+    if (v1parts.length != v2parts.length) {
+        return false;
+    }
 
-});
\ No newline at end of file
+    return 0;
+}
```

Comment:
You seem to repeat a lot 
```
_rStudioRequest('/rpc/console_input','console_input',urlParam("RS_SHARED_SECRET"),urlParam("Rstudio_port"),["Rdocumentation::makeDefault()"]);
``` 
in that file, better to wrap `_rStudioRequest` in another higher level function who just takes the command to send in argument